### PR TITLE
LPD-25341 , LPD-25349 Bug fixes for AutoSave feature

### PR DESCRIPTION
**Pull request by jkappler**
Story: [LPD-24020](https://liferay.atlassian.net/browse/LPD-24020)

Required Feature flags:
`feature.flag.LPD-11228=true
feature.flag.LPS-141392=true
feature.flag.LPD-15596=true`

Reproduction steps for [LPD-25341](https://liferay.atlassian.net/browse/LPD-25341):

1. Go to web content
2. Create a new web content
3. Add text on any of the fields and the title
4. Wait for it to be autosaved
5. Click on Publish

Expected behavior:
   There should only be a published version of the created content and no Draft versions of it
Current behavior
   The content is not published and 2 draft versions of the content are created

Reproduction steps for [LPD-25349](https://liferay.atlassian.net/browse/LPD-25349):

1. Go to web content
2. Create a new web content
3. Add text on any of the fields and the title
4. Wait for it to be autosaved
5. Navigate back to web contents without publishing
6. Reopen the web content for edit
7. edit the article
8. Hit Publish

Expected: The publish happens and the user is redirected to web contents list view

Actual: The publish happens, but the user is not redirected

[LPD-24020]: https://liferay.atlassian.net/browse/LPD-24020?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LPD-25341]: https://liferay.atlassian.net/browse/LPD-25341?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LPD-25349]: https://liferay.atlassian.net/browse/LPD-25349?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/JournalPortlet.es.js [hunk 1]
```diff
@@ -362,15 +362,19 @@ export default function _JournalPortlet({
...
+							).value = articleId;
 						}
 					}
-
-					lockHolder.lock?.unlock();
 				}
 			})
 			.catch((error) => {
 				console.error(error);
 				lockHolder.lock?.unlock();
```
Comment (ambrinchaudhary) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/JournalPortlet.es.js:374]: As we added a finally, I would say this is not necessary anymore

**Review by ambrinchaudhary — APPROVED** [review]
Just a minor sugestions, otherwise LGTM

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/JournalPortlet.es.js (hunk 1 above)
Comment (jkappler) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/JournalPortlet.es.js:374]: cc:// @NemethNorbert

Comment (NemethNorbert) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/JournalPortlet.es.js:374]: Yes we can remove`lockHolder.lock?.unlock();` from the catch block as well

**Comment by jkappler** [comment]
ci:forward

**Comment by jkappler** [comment]
thanks @NemethNorbert and @ambrinchaudhary!

**Comment by NemethNorbert** [comment]
ci:test:sf

**Comment by NemethNorbert** [comment]
ci:forward

**Pull request closed without merging**